src/Raven.Server/Documents/DocumentCompare.cs

```diff
@@ -14,14 +14,16 @@ public static class DocumentCompare
     {
         public readonly struct DocumentCompareOptions
         {
-            private DocumentCompareOptions(bool tryMergeMetadataConflicts, bool throwOnAttachmentModifications)
+            private DocumentCompareOptions(bool tryMergeMetadataConflicts, bool throwOnAttachmentModifications, bool compareDataArchivalMetadata = false)
             {
                 TryMergeMetadataConflicts = tryMergeMetadataConflicts;
                 ThrowOnAttachmentModifications = throwOnAttachmentModifications;
+                CompareDataArchivalMetadata = compareDataArchivalMetadata;
```

Comment:
Can you please elaborate more on this change?

Reply:
This introduces the possibility of enabling "@archive-at" metadata property check in `isEqual` method.
The design of Data Archival is that we don't create a new revision when "@archive-at" is added to the metadata (or when "@archived:true" appears). We have a test for this, which previously failed when this option wasn't introduced.

To prevent revision creation, we leave it default - `false`
We're setting it to `true` only in PatchDocumentCommand to detect actual changes to the document, so the patch test result contains the changed document. Otherwise, the patch test won't detect it, and the test result document won't contain the "@archived-at" flag.